### PR TITLE
feat: support new username system

### PR DESCRIPTION
**Pull request by aiko-chan-ai**
**Please describe the changes this PR makes and why it should be merged:**

- This PR adds support for the new username system (v13)
- #9512 & #9631 & #9660

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -2703,13 +2703,16 @@ export class User extends PartialTextBasedChannel(Base) {
   public readonly createdAt: Date;
   public readonly createdTimestamp: number;
   public discriminator: string;
+  public get displayName(): string;
```
Comment (jaw0r3k) [inline typings/index.d.ts:2706]:
```suggestion
  public readonly displayName: string;
```

File: src/util/Util.js [hunk 2]
```diff
@@ -741,6 +741,16 @@ class Util extends null {
       emoji_name: defaultReaction.name,
     };
   }
+
+  /**
+   * Calculates the default avatar index for a given user id.
+   * @param {Snowflake} userId - The user id to calculate the default avatar index for
+   * @returns {number}
+   * @ignore
```
Comment (jaw0r3k) [inline src/util/Util.js:749]:
```suggestion
```

in original pr this method isnt igonerd

Comment (aiko-chan-ai) [reply src/util/Util.js:749]: @jaw0r3k I think it should be ignored because we don't need this function in the documentation :/

Comment (jaw0r3k) [reply src/util/Util.js:749]: users may want to use it

Comment (aiko-chan-ai) [reply src/util/Util.js:749]: hmm...oke

File: typings/index.d.ts
Comment (Syjalo) [inline typings/index.d.ts:1]: Add typing for `calculateUserDefaultAvatarIndex`

**Review by Syjalo — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Comment by Jiralite** [comment]
Mind backporting #9660 too?

**Review by kyranet — APPROVED** [review]

File: src/structures/User.js [hunk 3]
```diff
@@ -193,12 +206,28 @@ class User extends Base {
   }
 
   /**
-   * The Discord "tag" (e.g. `hydrabolt#0001`) for this user
+   * The tag of this user
+   * <info>This user's username, or their legacy tag (e.g. `hydrabolt#0001`)
+   * if they're using the legacy username system</info>
    * @type {?string}
+   * @deprecated Use {@link User#username} instead.
```
Comment (Jiralite) [inline src/structures/User.js:213]: This still needs to be removed.

**Review by Jiralite — CHANGES_REQUESTED** [review]

**Review by Jiralite — APPROVED** [review]

**Pull request merged**